### PR TITLE
Fix a couple of promise bugs

### PR DESCRIPTION
**Pull request by bghgary**

### CODE REVIEW

File: tests/unit/babylon/src/Tools/babylon.promise.tests.ts [hunk 1]
```diff
@@ -45,13 +45,16 @@ describe('Babylon.Promise', function () {
             }).catch(() => {
                 tempString += " to check promises";
             }).then(() => {
-                expect(tempString).to.eq("Initial message to check promises");
-                done();
+                try {
+                    expect(tempString).to.eq("Initial message to check promises");
+                    done();
+                }
+                catch (error) {
```
Comment (deltakosh) [inline tests/unit/babylon/src/Tools/babylon.promise.tests.ts]: using a catch will not prevent error detection?
Why adding catch everywhere in these tests?

Comment (bghgary) [reply tests/unit/babylon/src/Tools/babylon.promise.tests.ts]: Async tests using done must call done with the error or the test will timeout.

Comment (bghgary) [reply tests/unit/babylon/src/Tools/babylon.promise.tests.ts]: Timeout when there is an error that is.

Comment (deltakosh) [reply tests/unit/babylon/src/Tools/babylon.promise.tests.ts]: ok :) thanks for the info

Comment (bghgary) [reply tests/unit/babylon/src/Tools/babylon.promise.tests.ts]: That's why promises are better for async unit tests, but we are testing promises in this test :)

**Pull request merged**